### PR TITLE
Earth - Sophie - Media Ranker Revisited

### PR DESCRIPTION
**Pull request by steve-messing**
# Assignment Submission: Media Ranker Revisited

Congratulations! You're submitting your assignment. Please reflect on the assignment with these questions.

## Reflection

Prompt | Response
--- | ---
What did you need to configure and setup to make OAuth installed and work? | add routes routes.rb, create .env with github client secret and client ID, add .env to .gitignore, add omniauth and dotenv-rails gems to Gemfile, create omniauth.rb initializer, create controller actions, create build_from_github method in user.rb model.
What areas of Rails app code did you need to create/write/modify in order to change logging in to use OAuth? | I had to modify some code in application.html.erb to make sure the buttons were routing to the correct paths.
What was one controller test you updated? Why did you need to update it? | I had to change post logout_path to delete logout_path to match the new Omniauth logout action.
Why did we need to mock OAuth for testing? | so that we don't provide sensitive data when we interact with the provider during tests.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+GITHUB_CLIENT_ID: 40215582fd61ce58b59b
+GITHUB_CLIENT_SECRET: 3900adb88d2eebecfccf8bf55ba3b69bfa44a79a
```
Comment (CheezItMan) [inline .env]: Hey an Oauth Token!  Hmmm... Now I can make a site masquerading as yours!

I see you added `.env` to your `.gitignore` file, but you need to remove this from your git history to get it to stop showing up.

File: app/controllers/works_controller.rb [hunk 2]
```diff
@@ -2,6 +2,26 @@ class WorksController < ApplicationController
...
+  end
+
+  def authorized?
+    unless @work.owner(@login_user)
+      flash[:status] = :failure
+      flash[:result_text] = "You are not authorized to do this."
+      redirect_back fallback_location: root_path
+      return
+    end
+  end
```
Comment (CheezItMan) [inline app/controllers/works_controller.rb:24]: Nice filter methods!

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -37,16 +37,16 @@
         <% if @login_user %>
 
         <li class="nav-item app-header__nav_item">
-          <%= link_to "Logged in as #{@login_user.username}", user_path(@login_user), class: "btn btn-primary" %>
+          <%= link_to "Logged in as #{@login_user.username}", user_path(@login_user), class: "btn btn-secondary" %>
         </li>
         <li class="nav-item app-header__nav_item">
-          <%= link_to "Log Out", logout_path, method: :post, class: "btn btn-primary" %>
+          <%= link_to "Log Out", logout_path, method: :delete, class: "btn btn-primary" %>
         </li>
```
Comment (CheezItMan) [inline app/views/layouts/application.html.erb:44]: 👍

**Review by CheezItMan — COMMENTED** [review]
Nice work Sophie, see my note on the .env file, but otherwise it works.  Do check out my tests, as they could have helped a bit ensuring you covered edge-cases.
